core/src/main/scala/kafka/server/RequestHandlerHelper.scala

```diff
@@ -58,16 +57,68 @@ object RequestHandlerHelper {
   }
 }
 
+class UnthrottledRequestHandlerHelper(
+  requestChannel: RequestChannel,
+  logPrefix: String
+) extends Logging {
+  this.logIdent = logPrefix
 
+  def handleError(request: RequestChannel.Request, e: Throwable): Unit = {
+    error("Error when handling request: " +
+      s"clientId=${request.header.clientId}, " +
+      s"correlationId=${request.header.correlationId}, " +
+      s"api=${request.header.apiKey}, " +
+      s"version=${request.header.apiVersion}, " +
+      s"body=${request.body[AbstractRequest]}", e)
+    sendErrorOrCloseConnection(request, e, 0)
```

Comment:
Did we consider having the throttle calls (`maybeRecordExempt`, `throttle`, `maybeRecordAndGetThrottleTimeMs`, etc.) split into a separate interface and then have an unthrottled implementation of that?

Reply:
The approach I took here was intended as a lighter touch. Many of the APIs exposed by `RequestHandlerHelper` set throttling expectations explicitly (e.g. `sendResponseExemptThrottle`). That makes it a little awkward to push down the throttling logic. If we only depended on the request quota, then I would be tempted to try, but we also depend on the controller mutation quota. I did consider changing the dependence from `QuotaManagers` to `ClientRequestQuotaManager` and `ControllerMutationQuotaManager`. Then we could pull out a trait for both of this. But all of this seemed like more work with little benefit.

Reply:
Thanks for the explanation. The `handleError` duplication seems a bit ugly. Also, inheriting from concrete classes is generally not great. So, to me, the current approach is lighter touch, but doesn't feel right.

Reply:
There is no state inside `UnthrottledRequestHandlerHelper`, so we could pull these methods into the companion I guess. Not sure that's really better, but at least it removes the inheritance. Let me see how this looks.